src/build.js

```diff
@@ -43,6 +45,74 @@ export async function generateBuildMap(configObj) {
   return await iterateDir(configObj.inputDir);
 }
 
+export async function generateBundleOutputObj(buildMap) {
+  const filenames = buildMap.map(buildPair => buildPair[0].pathname);
+
+  function PRESMPlugin() {
+    return {
+      name: 'PRESMPlugin',
+      resolveId(source) {
+        if (filenames.includes(source)) {
+          return source;
+        }
+        return null;
+      },
+      // If source is in buildmap, return source code
+      load(id) {
+        const sourceArr = buildMap.filter(
+          buildPair => buildPair[0].pathname === id
+        );
+        if (sourceArr) {
+          return sourceArr[0][1];
+        }
+        return null;
+      },
+    };
+  }
+
+  const inputOptions = {
+    input: filenames,
+    plugins: [PRESMPlugin()],
+  };
+
+  let preGenerate = true;
+
+  const outputOptions = {
+    dir: 'dist',
+    preserveModules: true,
+    // Determine extension for output files
+    entryFileNames: entry => {
+      if (preGenerate) {
+        return entry.name;
```

Comment:
When does this happen? May be good to document above.

Reply:
Added a quick explanation on when this `entryFileNames` hook is called in the workflow, right above this function.  Hopefully it clarifies things